src/NuGet.CommandLine.StagingExtensions/StageManagementResource.cs

```diff
@@ -142,6 +142,104 @@ public StageManagementResource(string stageServiceUri, HttpSource httpSource)
             return result;
         }
 
+        public async Task<StageDetailedView> Get(string id, Common.ILogger log)

```

Comment:
Do you think it might be better if `StageManagementResource` took in a `ILogger` upon creation and then proceeded to always log to it, rather than requiring to be passed a `ILogger` in each method? It seems like `StageCommand` gets a `StageManagementResource` fairly early on and then consistently passes `Console` to it in every method call, so I think it might look cleaner if we just gave it `Console` in a constructor. We could also store Console as an `ILogger` on `StageCommand` as well, and then replace every `Console` call with that so that we could easily switch the `ILogger` that `StageCommand` uses.